service/src/event/service.rs

```diff
@@ -197,225 +176,94 @@ impl CeramicEventService {
-        }
+        let to_insert = to_insert_with_metadata
+            .iter()
+            .map(|(e, _)| e.clone())
+            .collect::<Vec<_>>();
 
-        let deliverable =
-            OrderingState::discover_deliverable_events(pool, &mut to_check_map).await?;
-        if deliverable.len() != required_to_find - found_in_batch {
-            let missing = insert_if_greenlit
-                .values()
-                .filter_map(|(_, ev)| {
-                    if !deliverable.contains(&ev.body.cid) {
-                        Some(ev.body.cid)
-                    } else {
-                        None
-                    }
-                })
-                .collect::<Vec<_>>();
+        // need to make a better interface around events. needing to know about the stream in some places but
+        // not in the store makes it inconvenient to map back and forth
+        let res = CeramicOneEvent::insert_many(&self.pool, &to_insert[..]).await?;
 
-            tracing::info!(?missing, ?deliverable, "Missing required `prev` event CIDs");
+        // api writes shouldn't have any missed pieces that need ordering so we don't send those
+        if !history_required {
+            let to_send = res
+                .inserted
+                .iter()
+                .filter(|i| i.new_key)
+                .collect::<Vec<_>>();
 
-            Err(Error::new_invalid_arg(anyhow::anyhow!(
-                "Missing required `prev` event CIDs: {:?}",
-                missing
-            )))
-        } else {
-            // we need to use the deliverable list's order because we might depend on something in the same batch, and that function will
-            // ensure we have a queue in the correct order. So we follow the order and use our insert_if_greenlit map to get the details.
-            for cid in deliverable {
-                if let Some((meta, insertable)) = insert_if_greenlit.remove(&cid) {
-                    self.mark_event_deliverable_now(insertable, meta.init_cid);
+            for ev in to_send {
+                if let Some((ev, metadata)) = to_insert_with_metadata
+                    .iter()
+                    .find(|(i, _)| i.order_key == ev.order_key)
+                {
+                    let discovered = DiscoveredEvent {
+                        cid: ev.cid(),
+                        known_deliverable: ev.deliverable(),
+                        metadata: metadata.to_owned(),
+                    };
+                    trace!(?discovered, "sending delivered to ordering task");
+                    if let Err(e) = self.delivery_task.tx_inserted.try_send(discovered) {
+                        match e {
+                            tokio::sync::mpsc::error::TrySendError::Full(e) => {
+                                warn!(event=?e, limit=%PENDING_EVENTS_CHANNEL_DEPTH, "Delivery task full. Dropping event and will not be able to mark deliverable new stream event arrives or process is restarted");
```

Comment:
yeah, it might be worth just using send.. I increased the channel depth to 1 million (304 bytes each * 1M events ~= 3MB of data) for now which would mean we're way behind on processing...

[ticket AES-176](https://linear.app/3boxlabs/issue/AES-176/apply-backpressure-when-discovering-events-faster-than-order-task-can): 

```
Context: Currently, when we accept a new event and it's out of order, we send it to another task to process using a tokio mpsc channel and the try_send function. This fails immediately if the channel is full. We should consider whether blocking until it's available with send or for a time period using send_timeout is preferable. This would consume a task and may slow down writing new events while that's happening. 

We could also notify the task that we were unable to tell it of events at the time and store them in memory to retry when it's less congested, or tell it to seek them out in the database like it does at start up. 

We currently allow 1 million events in the channel buffer (3 MB of RAM), so we'd have to be way behind to encounter this.```

Reply:
actually, I just switched to send because slowing down is probably fine if we're that behind on pulling things out of the buffer. I didn't want to block writes before but this is probably a better default for now. I flipped the ticket to consider whether we should use try_send or send_timeout instead.